File: app/src/main/java/com/solt9029/editmasterandroid/viewmodel/ValiFieldFloat.java
package com.solt9029.editmasterandroid.viewmodel;

import androidx.annotation.Nullable;
import com.mlykotom.valifi.fields.number.ValiFieldNumber;

public class ValiFieldFloat extends ValiFieldNumber<Float> {
    public ValiFieldFloat() {
        super();
    }

    public ValiFieldFloat(@Nullable Float defaultValue) {
        super(defaultValue);
    }

    @Override
    protected Float parse(@Nullable String value) throws NumberFormatException {
        return Float.parseFloat(value);
    }
}

File: app/src/main/java/com/solt9029/editmasterandroid/viewmodel/ValiFieldFloat.kt
package com.solt9029.editmasterandroid.viewmodel

import com.mlykotom.valifi.fields.number.ValiFieldNumber

class ValiFieldFloat : ValiFieldNumber<Float> {
    constructor() : super() {}
    constructor(defaultValue: Float?) : super(defaultValue) {}

    @Throws(NumberFormatException::class)
    override fun parse(value: String?): Float? {
        return java.lang.Float.parseFloat(value!!)
    }
}
